/home/mwsrgf/act-venv/lib64/python3.9/site-packages/torch/__init__.py:955: UserWarning: torch.set_default_tensor_type() is deprecated as of PyTorch 2.1, please use torch.set_default_dtype() and torch.set_default_device() as alternatives. (Triggered internally at ../torch/csrc/tensor/python_tensor.cpp:432.)
  _C._set_default_tensor_type(t)
Warning: no DISPLAY environment variable.
--No graphics will be displayed.

        ACTSimulator (2024)
        ----------
        When submitting multiple jobs, note that the cells must share modfiles.
        
/home/mwsrgf/proj/ACT/experiments/LA_A/not-filtered/output/LAA_original-4/42/target/synthetic_0.04_0.045_0.05_0.055_0.06_0.0_0.1_0.2_0.3_0.4_4.5_5.0_5.5_6.0_6.5
im: (0.0013605732015421162,0.0023605732015421163) -- TRUE: 0.002
nap: (0.00014022423637919065,0.00021122423637919067) -- TRUE: 0.000142
na3: (0.02587456022446321,0.04087456022446321) -- TRUE: 0.03
kdr: (0.001332591946388383,0.002082591946388383) -- TRUE: 0.0015
cadyn: (3.790586357507963e-05,6.790586357507963e-05) -- TRUE: 6e-05
sAHP: (0.005954852306596899,0.010454852306596898) -- TRUE: 0.009
kap: (0.00046694631221240766,0.0008884463122124076) -- TRUE: 0.000843
-------------
Range (high - low) 
im range: 0.001 -- 50% TRUE: 0.001
nap range: 7.100000000000002e-05 -- 50% TRUE: 7.1e-05
na3 range: 0.015 -- 50% TRUE: 0.015
kdr range: 0.00075 -- 50% TRUE: 0.00075
cadyn range: 3.0000000000000004e-05 -- 50% TRUE: 3e-05
sAHP range: 0.004499999999999999 -- 50% TRUE: 0.0045
kap range: 0.00042149999999999995 -- 50% TRUE: 0.0004215
RUNNING THE MODULE
LOADING TARGET TRACES
SIMULATING TRAINING DATA

        ACTSimulator (2024)
        ----------
        When submitting multiple jobs, note that the cells must share modfiles.
        
Getting Conductance Combinations From Preselected Ranges
/home/mwsrgf/proj/ACT/experiments/LA_A/not-filtered/output/LAA_original-4/42/module_final/train
TRAINING RANDOM FOREST REGRESSOR
Extracting features: ['I_mean', 'I_stdev', 'Num Spikes', 'Spike Time 1', 'Spike Time 2', 'Spike Time 3', 'Spike Time 4', 'Spike Time 5', 'Spike Time 6', 'Spike Time 7', 'Spike Time 8', 'Spike Time 9', 'Spike Time 10', 'Spike Time 11', 'Spike Time 12', 'Spike Time 13', 'Spike Time 14', 'Spike Time 15', 'Spike Time 16', 'Spike Time 17', 'Spike Time 18', 'Spike Time 19', 'Spike Time 20', 'Min Spike Height', 'Max Spike Height', 'Avg Spike Height', 'Std Spike Height', 'Trough Time 1', 'Trough Time 2', 'Trough Time 3', 'Trough Time 4', 'Trough Time 5', 'Trough Time 6', 'Trough Time 7', 'Trough Time 8', 'Trough Time 9', 'Trough Time 10', 'Trough Time 11', 'Trough Time 12', 'Trough Time 13', 'Trough Time 14', 'Trough Time 15', 'Trough Time 16', 'Trough Time 17', 'Trough Time 18', 'Trough Time 19', 'Trough Time 20', 'Min Trough Height', 'Max Trough Height', 'Avg Trough Height', 'Std Trough Height', 'lto-hto_frequency', 'lto-hto_amplitude']
Evaluating random forest
joblib.externals.loky.process_executor._RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/home/mwsrgf/act-venv/lib64/python3.9/site-packages/joblib/externals/loky/process_executor.py", line 463, in _process_worker
    r = call_item()
  File "/home/mwsrgf/act-venv/lib64/python3.9/site-packages/joblib/externals/loky/process_executor.py", line 291, in __call__
    return self.fn(*self.args, **self.kwargs)
  File "/home/mwsrgf/act-venv/lib64/python3.9/site-packages/joblib/parallel.py", line 598, in __call__
    return [func(*args, **kwargs)
  File "/home/mwsrgf/act-venv/lib64/python3.9/site-packages/joblib/parallel.py", line 598, in <listcomp>
    return [func(*args, **kwargs)
  File "/home/mwsrgf/act-venv/lib64/python3.9/site-packages/sklearn/utils/parallel.py", line 136, in __call__
    return self.function(*args, **kwargs)
  File "/home/mwsrgf/act-venv/lib64/python3.9/site-packages/sklearn/model_selection/_validation.py", line 888, in _fit_and_score
    estimator.fit(X_train, y_train, **fit_params)
  File "/home/mwsrgf/act-venv/lib64/python3.9/site-packages/sklearn/base.py", line 1473, in wrapper
    return fit_method(estimator, *args, **kwargs)
  File "/home/mwsrgf/act-venv/lib64/python3.9/site-packages/sklearn/ensemble/_forest.py", line 489, in fit
  File "/home/mwsrgf/act-venv/lib64/python3.9/site-packages/sklearn/utils/parallel.py", line 74, in __call__
    return super().__call__(iterable_with_config)
  File "/home/mwsrgf/act-venv/lib64/python3.9/site-packages/joblib/parallel.py", line 1918, in __call__
    return output if self.return_generator else list(output)
  File "/home/mwsrgf/act-venv/lib64/python3.9/site-packages/joblib/parallel.py", line 1847, in _get_sequential_output
    res = func(*args, **kwargs)
  File "/home/mwsrgf/act-venv/lib64/python3.9/site-packages/sklearn/utils/parallel.py", line 136, in __call__
    return self.function(*args, **kwargs)
  File "/home/mwsrgf/act-venv/lib64/python3.9/site-packages/sklearn/ensemble/_forest.py", line 192, in _parallel_build_trees
    tree._fit(
  File "/home/mwsrgf/act-venv/lib64/python3.9/site-packages/sklearn/tree/_classes.py", line 472, in _fit
    builder.build(self.tree_, X, y, sample_weight, missing_values_in_feature_mask)
  File "_tree.pyx", line 172, in sklearn.tree._tree.DepthFirstTreeBuilder.build
  File "_tree.pyx", line 287, in sklearn.tree._tree.DepthFirstTreeBuilder.build
  File "_tree.pyx", line 942, in sklearn.tree._tree.Tree._add_node
  File "_tree.pyx", line 910, in sklearn.tree._tree.Tree._resize_c
  File "_utils.pyx", line 35, in sklearn.tree._utils.safe_realloc
MemoryError: could not allocate 524288 bytes
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/mwsrgf/proj/ACT/experiments/LA_A/not-filtered/Original.py", line 296, in <module>
    run()
  File "/home/mwsrgf/proj/ACT/experiments/LA_A/not-filtered/Original.py", line 168, in run
    predicted_g_data_file = mod.run()
  File "/home/mwsrgf/proj/ACT/act/ACTModule.py", line 55, in run
    prediction = self.get_rf_prediction()
  File "/home/mwsrgf/proj/ACT/act/ACTModule.py", line 367, in get_rf_prediction
    metrics.evaluate_random_forest(rf.model, 
  File "/home/mwsrgf/proj/ACT/act/Metrics.py", line 55, in evaluate_random_forest
    n_scores = abs(cross_val_score(
  File "/home/mwsrgf/act-venv/lib64/python3.9/site-packages/sklearn/utils/_param_validation.py", line 213, in wrapper
    return func(*args, **kwargs)
  File "/home/mwsrgf/act-venv/lib64/python3.9/site-packages/sklearn/model_selection/_validation.py", line 712, in cross_val_score
    cv_results = cross_validate(
  File "/home/mwsrgf/act-venv/lib64/python3.9/site-packages/sklearn/utils/_param_validation.py", line 213, in wrapper
    return func(*args, **kwargs)
  File "/home/mwsrgf/act-venv/lib64/python3.9/site-packages/sklearn/model_selection/_validation.py", line 423, in cross_validate
    results = parallel(
  File "/home/mwsrgf/act-venv/lib64/python3.9/site-packages/sklearn/utils/parallel.py", line 74, in __call__
    return super().__call__(iterable_with_config)
  File "/home/mwsrgf/act-venv/lib64/python3.9/site-packages/joblib/parallel.py", line 2007, in __call__
    return output if self.return_generator else list(output)
  File "/home/mwsrgf/act-venv/lib64/python3.9/site-packages/joblib/parallel.py", line 1650, in _get_outputs
    yield from self._retrieve()
  File "/home/mwsrgf/act-venv/lib64/python3.9/site-packages/joblib/parallel.py", line 1754, in _retrieve
    self._raise_error_fast()
  File "/home/mwsrgf/act-venv/lib64/python3.9/site-packages/joblib/parallel.py", line 1789, in _raise_error_fast
    error_job.get_result(self.timeout)
  File "/home/mwsrgf/act-venv/lib64/python3.9/site-packages/joblib/parallel.py", line 745, in get_result
    return self._return_or_raise()
  File "/home/mwsrgf/act-venv/lib64/python3.9/site-packages/joblib/parallel.py", line 763, in _return_or_raise
    raise self._result
MemoryError: could not allocate 524288 bytes
Start: Mon Jan 27 14:00:24 CST 2025 
End:   Mon Jan 27 15:28:02 CST 2025
Done running model at Mon Jan 27 15:28:02 CST 2025
